server/automute.go

```diff
@@ -151,24 +141,24 @@ func (p *Plugin) canAutomuteChannelID(channelID string) (bool, error) {
 	return p.canAutomuteChannel(channel)
 }
 
-// canAutomuteChannel returns true if the channel is either explicitly linked to a channel in MS Teams or if it's a
-// DM/GM channel that is implicitly linked to MS Teams.
+// canAutomuteChannel returns true if the channel is linked to a channel in MS Teams
+// DM/GM channels are muted depending on Selective Sync and Sync Remote Only settings
 func (p *Plugin) canAutomuteChannel(channel *model.Channel) (bool, error) {
-	// Automute all GM channels
-	if channel.Type == model.ChannelTypeGroup {
-		return true, nil
-	} else if channel.Type == model.ChannelTypeDirect {
-		userIDs := strings.Split(channel.Name, "__")
-		for _, userID := range userIDs {
-			user, appErr := p.API.GetUser(userID)
-			if appErr != nil {
-				return false, errors.Wrap(appErr, fmt.Sprintf("Unable to get user for channel member %s ", userID))
-			}
-			if user.IsBot || user.IsGuest() {
+	if channel.IsGroupOrDirect() {
+		if p.configuration.SelectiveSync {
+			if p.configuration.SyncRemoteOnly {
+				// if only sync'ing with remote users,
+				// do not automute any DM/GM channels,
+				// in this mode all uses considered MM primary
 				return false, nil
 			}
+			// if Selective Sync, automute if members
+			// span platforms
+			return p.ChannelShouldSync(channel.Id)
 		}
-		return true, nil
+
+		// if not selective sync
+		return false, nil
 	}
 
 	link, err := p.store.GetLinkByChannelID(channel.Id)
```

Comment:
Should we check `p.configuration.SelectiveSync` at the top of this method? Otherwise, it looks like we can returne `true` here for linked channels even when selective sync is false.

Reply:
@lieut-data  Sorry, I don't understand this comment. `p.configuration.SelectiveSync` is checked above if a GM or DM. And previous comment was to not to automute unless SS was enabled. 

> Commented above, but I thought we didn't automute /unless/ selective sync was enabled. So maybe we can simplify?

I did not think "linked channels" had anything to do with Selective sync. The logic around linked channels has not changed.

Reply:
My mistake: I think I conflated the automute/Teams primary functionality with selective sync, but the latter came after. This is definitely ripe for consolidation -- discard my feedback for now, and we can revisit later.